tests/test_core/test_alignment.py

```diff
@@ -1530,6 +1530,27 @@ class AlignmentBaseTests(SequenceCollectionBaseTests):
     as a constructor.
     """
 
+    def test_alignment_quality(self):
+        """Tests that the alignment_quality generates the right alignment quality
+        value based on the Hertz-Stormo metric."""
+        aln = make_aligned_seqs(["AATTGA",
+                                 "AGGTCC",
+                                 "AGGATG",
+                                 "AGGCGT"], moltype="dna")
+        got = aln.alignment_quality(equifreq_mprobs=True)
+        expect = log2(4) + (3 / 2) * log2(3) + (1 / 2) * log2(2) + (1 / 2) * log2(2)
+        assert_allclose(got, expect)
+
+        aln = make_aligned_seqs(["AAAC",
+                                 "ACGC",
+                                 "AGCC",
+                                 "A-TC"], moltype="dna")
+        got = aln.alignment_quality(equifreq_mprobs=False)
+        expect = 2 * log2(1 / 0.4) + log2(1 / (4 * 0.4)) + (1 / 2) * log2(1 / (8 / 15)) + (
+                1 / 4) * log2(1 / (4 / 15))
+        assert_allclose(got, expect)
+
```

Comment:
1. test when `alignment' is all gaps
2. when alignment has a single sequence
3. when alignment has two seqs, but one is all gaps

2 and 3 should evaluate when equifreq True or False